packages/dds/quorum/.eslintrc.js

```diff
@@ -5,10 +5,25 @@
 
 module.exports = {
     "extends": [
-        "@fluidframework/eslint-config-fluid/strict",
+        require.resolve("@fluidframework/eslint-config-fluid/strict"),
```

Comment:
this seems like an unrelated change. Also, why do we want to use require.resolve here, but not for "prettier"?



There are several config files that do it the way this one does (my regex suggests there are 17 such configs).
Maybe the simpler version without require.resolve works?

My understanding of what require.resolve does suggests that using it in a top level config file thats only used as part of the project its located in makes it pointless since it shouldn't always work fine without it.

Reply:
eslint's default module resolution results in transitive deps not beingrespected. I.e. for a config extension chain like `a -(extends)->b->c`, `a` is required to directly depend on `c`. This applies to all "extends" and "plugin" entries. Directly `require`ing `b` from `a` lets node resolve the dependencies, and ends up yielding the behavior you would generally expect from a node tool.

This pattern is a pretty widely used workaround. I'll have to see if I can dig up the github issue on this.

It's possible that hoisting the require calls up into our shared config might also work, and would remove the requirement on the consumer side. I would have to test it.

Either way, these changes could be pulled into a separate PR, so that's probably the best call.

Reply:
This PR has more details (and is where these changes were initially made): #10410

Reply:
My understanding is the shared config should do `require.resolve` because it might refer to packages which don't get hoisted into the root, and thus won't be found without it, unless you get lucky and npm happened to hoist those deps.

In general, code which is distributed with a package, and needs to generate string paths to packages which it passes off to some other library should use require.resolve instead of unqualified strings.

Doing it in this file, which is not distributed as part of its package seems pointless. This file is always used when running eslint inside this packages' directory, and since this package declares a dependency on the referenced package, the package manager (npm for now) will ensure this path is in scope for the only case that its used.

Reply:
https://github.com/eslint/eslint/issues/3458#issuecomment-132922673 notes that it's the shred config that should use module objects. This justification does not apply to non-shared configs like this one. I think this approach was misapplied in your previous change you linked.

Reply:
Yeah, I think that is the better approach. At this point I don't remember why it wasn't the approach I ended up taking. I'll have to do some testing.

Note that this pattern did allow us to remove the dependencies on transitive configs, so it does end up having an impact in at least some cases, but I'll have to see if I can still repro. 

I have removed this change from the PR, though, so I'm going to resolve this issue for now. I'll keep exploring though :)